core/app/models/spree/address.rb

```diff
@@ -191,7 +191,8 @@ def state_validate
       # ensure state_name belongs to country without states, or that it matches a predefined state name/abbr
       if state_name.present?
         if country.states.present?
-          states = country.states.find_all_by_name_or_abbr(state_name)
+          state_lookup = state_name.downcase.strip
+          states = country.states.where('lower(name) = ? or lower(abbr) = ?', state_lookup, state_lookup);
```

Comment:
`matches` will use the appropriate case insensitive matcher depending on the DB. For example on Postgres:

```Ruby
Spree::State.where(Spree::State.arel_table[:abbr].matches('NY'))
#=>   Spree::State Load (2.9ms)  SELECT "spree_states".* FROM "spree_states" WHERE ("spree_states"."abbr" ILIKE 'NY')
```